src/Bicep.Core/Semantics/SemanticModel.cs

```diff
@@ -188,10 +189,33 @@ public IEnumerable<IDiagnostic> GetAllDiagnostics()
 
         private IEnumerable<IDiagnostic> AssembleDiagnostics()
```

Comment:
Would be worth changing the return type of this function to `IReadOnlyList<IDiagnostic>`. It looks like the existing code actually already has a bug where the enumeration can happen multiple times.

Reply:
Updated as part of this commit:
https://github.com/Azure/bicep/commit/7f012613ae1aad9fd2e8101ac82b73b362b8686b